ml/jml/feature_set.h

```diff
@@ -365,6 +365,9 @@ class Mutable_Feature_Set : public Feature_Set {
     virtual std::tuple<const Feature *, const float *, int, int, size_t>
     get_data(bool need_sorted = false) const
     {
+        if(features.size() == 0)
+            throw ML::Exception("features_type is empty!");
```

Comment:
I put that in because 2 lines lower, it tries to access the first element of features and so we're vulnerable to a segfault without that check